apps/server/src/modules/h5p-library-management/service/h5p-library-management.service.ts

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+import {
+	H5PConfig,
+	cacheImplementations,
+	LibraryManager,
+	ContentTypeCache,
+	IUser,
+	LibraryAdministration,
+} from '@lumieducation/h5p-server';
+import ContentManager from '@lumieducation/h5p-server/build/src/ContentManager';
+import ContentTypeInformationRepository from '@lumieducation/h5p-server/build/src/ContentTypeInformationRepository';
+import { Injectable } from '@nestjs/common';
+import { ContentStorage, LibraryStorage } from '@src/modules/h5p-editor/service';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { readFileSync } from 'fs';
+import { parse } from 'yaml';
+
+const h5pConfig = new H5PConfig(undefined, {
+	baseUrl: '/api/v3/h5p-editor',
+	contentUserStateSaveInterval: false,
+	setFinishedEnabled: false,
+});
+
+@Injectable()
+export class H5PLibraryManagementService {
+	contentTypeCache: ContentTypeCache;
+
+	contentTypeRepo: ContentTypeInformationRepository;
+
+	libraryManager: LibraryManager;
+
+	libraryAdministration: LibraryAdministration;
+
+	libraryWishList: string[];
+
+	constructor(private readonly libraryStorage: LibraryStorage, private readonly contentStorage: ContentStorage) {
+		const kvCache = new cacheImplementations.CachedKeyValueStorage('kvcache');
+		this.contentTypeCache = new ContentTypeCache(h5pConfig, kvCache);
+		this.libraryManager = new LibraryManager(
+			this.libraryStorage,
+			undefined,
+			undefined,
+			undefined,
+			undefined,
+			undefined,
+			h5pConfig
+		);
+		this.contentTypeRepo = new ContentTypeInformationRepository(this.contentTypeCache, this.libraryManager, h5pConfig);
+		const contentManager = new ContentManager(this.contentStorage);
+		this.libraryAdministration = new LibraryAdministration(this.libraryManager, contentManager);
+		const librariesYamlContent = readFileSync(Configuration.get('H5P_EDITOR__LIBRARY_LIST_PATH') as string, {
+			encoding: 'utf-8',
+		});
+		this.libraryWishList = (parse(librariesYamlContent) as { h5p_libraries: string[] }).h5p_libraries;
+	}
+
+	async uninstallUnwantedLibraries(wantedLibraries: string[]) {
+		for (const installedLibrary of await this.libraryAdministration.getLibraries()) {
+			if (!wantedLibraries.includes(installedLibrary.machineName) && installedLibrary.dependentsCount === 0) {
+				// force removal, don't let content prevent it, therefore use libraryStorage directly
+				// also to avoid conflicts, remove one-by-one, not using for-await:
+				// eslint-disable-next-line no-await-in-loop
+				await this.libraryStorage.deleteLibrary(installedLibrary);
```

Comment:
Is not allowed and we do not want to allow it.

Reply:
solved: 10d27f0a5c492efcc1f753a6153684c6d414ecc0